grovedb/src/replication.rs

```diff
@@ -31,6 +31,18 @@ struct SubtreeStateSyncInfo<'db> {
     num_processed_chunks: usize,
 }
 
+impl SubtreeStateSyncInfo<'_> {
+    // Function to create an instance of SubtreeStateSyncInfo with default values
+    fn new() -> Self {
+        let pending_chunks = BTreeSet::new();
+        Self {
+            restorer: None,
+            pending_chunks,
+            num_processed_chunks: 0,
+        }
+    }
```

Comment:
You should have SubtreeStateSyncInfo derive Default, and then just do Self::default() here for new.